scripts/backend.py

```diff
@@ -169,6 +238,312 @@ def page_not_found(error):
+
+@app.route("/admin", methods=['GET', 'POST'])
+@login_required
+def admin_page():
+    if (current_user.id != 'admin'):
+        return redirect("/")
+    if request.method == 'GET':
+        user_ban_info = db.get_ban_list()
+        return render_template("admin.html", all_users=user_ban_info)
+    else:
+        username = request.form['username']
+        new_user = db.make_user(username)
+        if new_user is None:
+            return 'error, {} already exists'.format(username)
+        else:
+            return 'success, {} created, their password is {}'.format(*new_user)
+
+
+@app.route("/dj")
+@login_required
+def dj_home_page():
+    dj_info = db.get_dj_info(current_user.id)
+    djin = [('your display name', 'dj_name'), ('dj pic url (optional)', 'dj_pic'), ('stream title (optional)', 'stream_title')]
+    if dj_info is None:
+        return 'something went wrong'
+    return render_template("dj_page.html", livestream_info=LIVESTREAM_INFO,
+                           dj_info_dict=dj_info, dj_info_names=djin,
+                           current_desc=dj_info['stream_desc'])
+
+
+@app.route("/dj/edit_dj_info", methods=['GET', 'POST'])
+@login_required
+def dj_edit_page():
+    if request.method == 'GET':
+        dj_info = db.get_dj_info(current_user.id)
+        djin = [('your display name', 'dj_name'), ('dj pic url (optional)', 'dj_pic'), ('edit your stream title (optional)', 'stream_title')]
+        if dj_info is None:
+            return 'something went wrong'
+        return render_template("dj_page_edit.html", livestream_info=LIVESTREAM_INFO,
+                               dj_info_dict=dj_info, dj_info_names=djin,
+                               current_desc=dj_info['stream_desc'])
+    else:
+        did_it_work = db.update_dj_info(current_user.id, request.form.to_dict())
+        if did_it_work:
+            return redirect('/dj')
+        else:
+            return 'whoops, something went wrong'
+
+
+@app.route("/dj/password_change_form", methods=['GET', 'POST'])
+@login_required
+def change_pass_page():
+    if request.method == 'GET':
+        return render_template("change_password.html")
+    else:
+        current_pass = request.form['current_pass']
+        new_pass = request.form['new_pass']
+        double_check = request.form['double_check']
+        if current_user.password == current_pass:
+            if new_pass == double_check:
+                db.change_password(current_user.id, new_pass)
+                logout_user()
+                return 'successfully changed your password, you\'ll need to log back in'
+        return redirect('/dj/password_change_form')
+
+
+@app.route("/admin/ban/<username>")
+@login_required
+def ban_user(username=None):
+    if (current_user.id != 'admin'):
+        return redirect("/")
+    if username is not None:
+        ban_result = db.ban_user(username, True)
+        if ban_result is not None:
+            return '{} is now banned'.format(username)
+    return 'nothing happened'
```

Comment:
Probably better to give a more meaningful error message about the user not being found.